client/lib/domains/is-subdomain.ts

```diff
@@ -1,14 +1,69 @@
 import { getRootDomain } from 'calypso/lib/domains/utils';
 
-export function isSubdomain( domainName: string ): boolean {
+export function isSubdomain( domainName: string, tld?: string ): boolean {
 	if ( ! domainName || domainName === '' ) {
 		return false;
 	}
+
 	const isValidSubdomain = Boolean(
 		domainName.match(
 			/^([a-z0-9_]([a-z0-9\-_]*[a-z0-9_])?\.)+[a-z0-9]([a-z0-9-]*[a-z0-9])?\.[a-z]{2,63}$/
 		)
 	);
 
-	return isValidSubdomain && getRootDomain( domainName ) !== domainName;
+	// If it doesn't even match the subdomain "shape", just return false.
```

Comment:
Have you considered passing down the `tld` param to `getRootDomain` and then to `getTld`? It seems it would make the code significantly simpler while keeping backward compatability for those methods if they're called without the `tld` param

Reply:
Ramy also suggested an approach where I have the backend return 'is_subdomain': boolean instead of using TLD at all. I think that seems like a cleaner approach overall, do you agree?

Reply:
p1738004177599369-slack-C07GK8PBRT9